e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package tests
+
+import (
+	"fmt"
+	"net/netip"
+	"os"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	frrk8sv1beta1 "github.com/metallb/frr-k8s/api/v1beta1"
+	"github.com/metallb/frrk8stests/pkg/config"
+	"github.com/metallb/frrk8stests/pkg/dump"
+	"github.com/metallb/frrk8stests/pkg/infra"
+	"github.com/metallb/frrk8stests/pkg/k8s"
+	"github.com/metallb/frrk8stests/pkg/k8sclient"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.universe.tf/e2etest/pkg/container"
+	"go.universe.tf/e2etest/pkg/executor"
+	"go.universe.tf/e2etest/pkg/frr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+)
+
+var FRRImage string
+
+// The Unnumbered test can run on IPv4, IPv6 or Dual because it needs only the
+// IPv6 LLA addresses in the interface, not full IPv6 stack.
+var _ = ginkgo.Describe("Unnumbered", func() {
+	var (
+		nodeWithP2PConnection corev1.Node
+		externalP2PContainer  *frrcontainer.FRR
+		cs                    clientset.Interface
+		updater               *config.Updater
+	)
+
+	ginkgo.BeforeEach(func() {
+		if _, found := os.LookupEnv("RUN_FRR_CONTAINER_ON_HOST_NETWORK"); found {
+			ginkgo.Skip("Skipping this test because RUN_FRR_CONTAINER_ON_HOST_NETWORK is set to true")
+		}
+		var err error
+		updater, err = config.NewUpdater()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = updater.Clean()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		nodes, err := k8s.Nodes(cs)
+		Expect(err).NotTo(HaveOccurred())
+		nodeWithP2PConnection = nodes[0]
+
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			reporter := dump.NewK8sReporter(k8s.FRRK8sNamespace)
+			testName := ginkgo.CurrentSpecReport().FullText()
+			dump.K8sInfo(testName, reporter)
+			dump.BGPInfo(testName, []*frrcontainer.FRR{externalP2PContainer}, cs)
+		}
+	})
+
+	ginkgo.DescribeTable("BGP session is established and routes are verified", func(rc frrconfig.RouterConfigUnnumbered) {
+		iface := rc.Interface
+		remoteASN := rc.ASNLocal
+		prefixSend := []string{"5.5.5.0/24", "5555::/64"}
+
+		externalP2PContainer, err := frrcontainer.CreateP2PPeerFor(nodeWithP2PConnection.Name, iface, FRRImage)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.By(fmt.Sprintf("create p2p %s:%s -- %s:%s)", nodeWithP2PConnection.Name, iface, iface, externalP2PContainer.Name))
+		ginkgo.DeferCleanup(func() {
+			err := frrcontainer.Delete([]*frrcontainer.FRR{externalP2PContainer})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		ginkgo.By(fmt.Sprintf("load frrconfig to %s", externalP2PContainer.Name))
```

Comment:
nit: -> `updating frr config for container %s`

Reply:
done